components/button/floating-buttons.js

```diff
@@ -147,75 +165,98 @@ class FloatingButtons extends RtlMixin(LitElement) {
 		`;
 	}
 
-	_calcContainerPosition() {
-		this._floating = this._shouldFloat();
-		if (!this._floating || !this.offsetParent) {
-			return;
+	updated(changedProperties) {
+		if (changedProperties.has('alwaysFloat')) {
+			this._recalculateFloating();
 		}
+	}
 
-		const offsetParentBoundingRect = this.offsetParent.getBoundingClientRect();
-		const boundingRect = this.getBoundingClientRect();
+	async _calcContainerPosition() {
+		requestAnimationFrame(() => {
 
-		const offsetParentLeft = offsetParentBoundingRect.left;
-		const left = boundingRect.left;
-		const containerLeft = left - offsetParentLeft - 1;
-		if (containerLeft !== 0) {
-			// only update this if needed - needed for firefox
-			this._containerMarginLeft = `-${containerLeft}px`;
-		}
+			this._containerMarginLeft = '';
+			this._containerMarginRight = '';
+			this._innerContainerLeft = '';
+			this._innerContainerRight = '';
```

Comment:
We weren't resetting these before, which left the margins on the containers even when things weren't floating.